prql-compiler/src/ast/pl/stmt.rs

```diff
@@ -64,6 +71,7 @@ pub struct FuncParam {
 pub struct VarDef {
     pub name: String,
     pub value: Box<Expr>,
+    pub ty_expr: Option<Expr>,
```

Comment:
Yes, nice.

But we are not parsing this yet, right?

Reply:
Wait, we don't need to parse this yet in this PR. Functions will be parsed from this:

```
let a = p -> b
```
into:
```yaml
VarDef:
- name: a,
- value:
    FuncDef:
      positional_params:
      - p
      body:
      - b
```

Now, `FuncDef` can easily be converted into `Closure` in `resolver::closure_of_var_def`.

Reply:
Wait, `ExprKind::FuncDef` and `ExprKind::Closure` are now approximately the same thing! 

You don't even need to add `ExprKind::FuncDef` and just parse into `ExprKind::Closure`, which can be renamed into `ExprKind::LambdaFunc` or just `ExprKind::Func`?

Reply:
> You don't even need to add `ExprKind::FuncDef` and just parse into `ExprKind::Closure`, which can be renamed into `ExprKind::LambdaFunc` or just `ExprKind::Func`?

Super! That's a great train-of-thought :)